### PR TITLE
Cloudwatch alarm support

### PR DESCRIPTION
**Pull request by eredi93**
i refactored https://github.com/dtan4/terraforming/pull/182 as i saw no activity on it.
Hope you can merge it in 😄

### CODE REVIEW

File: lib/terraforming/resource/cloud_watch_alarm.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      def module_name_of(alarm)
+        normalize_module_name(alarm.alarm_name)
+      end
+
+      def sanitize(argument)
+        argument.nil? ? "" : argument
+      end
+
+      def add_checksummed_attributes(attributes, alarm)
+        ["insufficient_data_actions", "alarm_actions", "ok_actions", "dimensions"].each do |action|
```
Comment (dtan4) [inline lib/terraforming/resource/cloud_watch_alarm.rb]: __[rubocop]__
* Use `%w` or `%W` for an array of words. [:ref](https://github.com/bbatsov/ruby-style-guide#percent-w)

File: lib/terraforming/resource/cloud_watch_alarm.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          attribute = alarm.send(action.to_sym)
+          attributes["#{action}.#"] = attribute.size.to_s
+          attribute.each do |attr|
+            if attr.is_a? String
+              checksum = Zlib.crc32(attr)
+              value = attr
```
Comment (dtan4) [inline lib/terraforming/resource/cloud_watch_alarm.rb]: __[rubocop]__
* Trailing whitespace detected. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-whitespace)

File: spec/lib/terraforming/resource/cloud_watch_alarm_spec.rb [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+      end
+
+      let(:alarms) do
+        [
+          {
+            actions_enabled: true,
+            alarm_actions: ["arn:aws:sns:region:account:lambda-alerts"],
+            alarm_name: "Alarm With Dimensions",
+            comparison_operator: "GreaterThanOrEqualToThreshold",
+            dimensions: [ {name: "FunctionName", value: "beep-beep"} ],
```
Comment (dtan4) [inline spec/lib/terraforming/resource/cloud_watch_alarm_spec.rb]: __[rubocop]__
* Space inside square brackets detected. [:ref](https://github.com/bbatsov/ruby-style-guide#no-spaces-braces)
* Space inside { missing. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-operators)
* Space inside } missing. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-operators)

**Comment by dtan4** [comment]
LGTM 👍  Thanks ❗

**Comment by eredi93** [comment]
thanks :hand:

**Pull request merged**